airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,546 @@
+            kubernetes_role = self.connection.extra_dejson.get("kubernetes_role")
+        return kubernetes_jwt_path, kubernetes_role
+
+    def _get_gcp_parameters_from_connection(
+        self, gcp_key_path: Optional[str], gcp_scopes: Optional[str]) \
+            -> Tuple[Optional[str], Optional[str]]:
+        if not gcp_scopes:
+            gcp_scopes = self.connection.extra_dejson.get("gcp_scopes")
+        if not gcp_key_path:
+            gcp_key_path = self.connection.extra_dejson.get("gcp_key_path")
+        return gcp_key_path, gcp_scopes
+
+    def get_conn(self) -> hvac.Client:
+        """
+        Retrieves connection to Vault.
+
+        :rtype: hvac.Client
+        :return: connection used.
+        """
+        return self.vault_client.client
+
+    def get_secret(self, secret_path: str, secret_version: Optional[int] = None) -> Optional[dict]:
+        """
+        Get secret value from the engine.
+
+        :param secret_path: Path of the secret
+        :type secret_path: str
+        :param secret_version: Optional version of key to read - can only be used in case of version 2 of KV
+        :type secret_version: int
+
+        See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v1.html
+        and https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        :param secret_path: Path of the secret
+        :type secret_path: str
+        :rtype: dict
+        :return: secret stored in the vault as a dictionary
+        """
+        return self.vault_client.get_secret(secret_path=secret_path, secret_version=secret_version)
+
+    def get_secret_metadata(self, secret_path: str) -> Optional[dict]:
+        """
+        Reads secret metadata (including versions) from the engine. It is only valid for KV version 2.
+
+        :param secret_path: Path to read from
+        :type secret_path: str
+        :rtype: dict
+        :return: secret metadata. This is a Dict containing metadata for the secret.
+
+                 See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        """
+        return self.vault_client.get_secret_metadata(secret_path=secret_path)
+
+    def get_secret_including_metadata(self,
+                                      secret_path: str,
+                                      secret_version: Optional[int] = None) -> Optional[dict]:
+        """
+        Reads secret including metadata. It is only valid for KV version 2.
+
+        See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        :param secret_path: Path of the secret
+        :type secret_path: str
+        :param secret_version: Optional version of key to read - can only be used in case of version 2 of KV
+        :type secret_version: int
+        :rtype: dict
+        :return: key info. This is a Dict with "data" mapping keeping secret
+                 and "metadata" mapping keeping metadata of the secret.
```

Comment:
```suggestion
        :return: key info. This is a Dict with "data" mapping keeping secret
            and "metadata" mapping keeping metadata of the secret.
```

I think is the correct indentation?

Reply:
Corrected and will double check.